manifests/params.pp

```diff
@@ -269,12 +269,16 @@
   $dhcp_load_balance = 3
 
   # DNS settings - requires optional DNS puppet module
+  if fact('networking.domain') {
+    $dns_zone = $facts['networking']['domain']
+  } else {
+    $dns_zone = ''
```

Comment:
I think this isn't correct. In Puppet you can use
```puppet
if $dns_zone {
  # Do something
} else {
  fail()
}
```

Back in Puppet 3 and earlier it would handle empty strings equal to `undef`. However, since Puppet 4 the empty string evaluates to true. Because of this, empty strings are not recommended -- too easy to have logic errors.

That's why I'd prefer `undef`. It may mean we need to change some types, but otherwise you get odd compilation errors in proxydns. It's a good thing to tighten them. In particular proxydns.pp could use some.